test/common/3Scale_user_promotion.go

```diff
@@ -0,0 +1,92 @@
+package common
+
+import (
+	goctx "context"
+	"fmt"
+	"github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/test/resources"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"testing"
+	"time"
+)
+
+func Test3ScaleUserPromotion(t *testing.T, ctx *TestingContext) {
+
+	var (
+		developerUser      = fmt.Sprintf("%v-%d", DefaultTestUserName, 0)
+		dedicatedAdminUser = fmt.Sprintf("%v-%d", defaultDedicatedAdminName, 0)
+	)
+
+	if err := createTestingIDP(t, goctx.TODO(), ctx.Client, ctx.KubeConfig, ctx.SelfSignedCerts); err != nil {
+		t.Fatalf("error while creating testing idp: %v", err)
+	}
+
+	// get console master url
+	rhmi, err := getRHMI(ctx.Client)
+	if err != nil {
+		t.Fatalf("error getting RHMI CR: %v", err)
+	}
+
+	// Get the 3Scale host url from the rhmi status
+	host := rhmi.Status.Stages[v1alpha1.ProductsStage].Products[v1alpha1.Product3Scale].Host
```

Comment:
e2e tests seems to be failing here as the Host value was unable to be retrieved - probably worth failing early here if the host value can't be retrieved/ is blank